### PR TITLE
lesson 1.6 Reusable components, Imperative React

### PR DESCRIPTION
**Pull request by AE-BHH**
Lesson_1_ 6 assignment;
Reusable components
Imperative React

### CODE REVIEW

File: src/AddTodoForm.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import React from 'react'
+import InputWithLabel from './InputWithLabel'
+
+function AddTodoForm(props) {
+	const { onAddTodo } = props
```
Comment (JoHeyJo) [inline src/AddTodoForm.js:5]: Is it necessary to destructure props with this additional line of code?

File: src/InputWithLabel.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+function InputWithLabel({ todoTitle, handleTitleChange, children }) {
+	const inputRef = useRef()
+
+	useEffect(() => {
+		inputRef.current.focus()
+	})
+	return (
+		<>
+			<label htmlFor='todoTitle'>{children}</label>
```
Comment (JoHeyJo) [inline src/InputWithLabel.js:12]: Maintain abstract naming conventions when naming attributes in an abstract component.

File: src/TodoListItem.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+
+function TodoListItem({ listItem, onRemoveTodo }) {
+	console.log(listItem)
+	return (
+		<li>
+			{listItem.title}
+			<button type='button' onClick={() => onRemoveTodo(listItem.id)}>
```
Comment (JoHeyJo) [inline src/TodoListItem.js:8]: Nice job applying this oneliner!

File: src/App.js [hunk 4]
```diff
@@ -1,18 +1,37 @@
...
-        <a href="https://reactjs.org" target="_blank" rel="noopener noreferrer">
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+	const [todoList, setTodoList] = useSemiPersistentState('savedTodoList', [])
+
+	function removeTodo(id) {
+		const updatedTodoListArr = todoList.filter((item) => item.id !== id)
```
Comment (JoHeyJo) [inline src/App.js:20]: Nice job building this function! Consider removing "Arr" from your variable name. It's not necessary to specify what kind of data structure is contained within a variable. Given good variable naming this kind of information should be inferred from the name.
By naming something "list" it's implied that it's a collection of things that naturally belong in an array. The same goes for using a "plural" naming convention. e.g. `todos` implies a list of todos as opposed to `todo` which would be a singular object.

File: src/AddTodoForm.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+	function handleAddTodo(event) {
+		event.preventDefault()
+
+		onAddTodo({ title: todoTitle, id: Date.now() })
+		setTodoTitle('')
+	}
+
+	return (
+		<>
```
Comment (JoHeyJo) [inline src/AddTodoForm.js:21]: You have an interesting bug in your user interface. This is something that affects the user experience. Play around with your app and see if you notice anything that's not working as expected. Don't spend too much time trying to find the bug so lmk if you spend more than 5 minutes on it. I'd rather you spend time fixing it. Good luck!

**Review by JoHeyJo — CHANGES_REQUESTED** [review]
Fantastic job getting your application to work! 
There are a few improvements and some code clean-up that needs to be done before merging. Overall, though you did a great job!

File: src/AddTodoForm.js (hunk 1 above)
Comment (AE-BHH) [reply src/AddTodoForm.js:5]: I guess no though it was on the text  book that was describing how it works. I will clean up!

Comment (JoHeyJo) [reply src/AddTodoForm.js:5]: Nice job taking direction from the text and understanding how to apply it but now let's focus on maintaining a consistent coding pattern. e.g. your other components destructure props in the component signature while this one doesn't. You've already set a pattern to follow. Also, it's more common to destructure in the component signature than creating an additional line of code. That type of coding pattern actually screams "junior"! :)

**Pull request merged**